### PR TITLE
fix(picker)!: adjust position of quiet picker with side label 

### PR DESCRIPTION
**Pull request by jenndiaz**
## Description

BREAKING CHANGE - does require a change to the classes used on the field label 

Addresses concerned raised in [SWC Github issue 2963]( https://github.com/adobe/spectrum-web-components/issues/2963) where the quiet picker and it's associated side label does not align with text of other components, making it appear incorrect when used with other components. 

For before look at #2231 

### Proposed fix: 
Adjust the top padding or margin for both the FieldLabel and Picker when the Picker is Quiet and has a side label.
#### Field Label

- add a class to the Field Label for this particular use case `spectrum-FieldLabel-quiet-picker-side-label`
- this class to position the field label text using custom properties: 
`--spectrum-fieldlabel-picker-side-label-quiet-padding-block-start-small`
`--spectrum-fieldlabel-picker-side-label-quiet-padding-block-start-medium`
`--spectrum-fieldlabel-picker-side-label-quiet-padding-block-start-large` 
`--spectrum-fieldlabel-picker-side-label-quiet-padding-block-start-xlarge`

#### Picker 
- using the already in place `spectrum-Picker--sideLabel`  class position the Picker using the custom properties: 
  `--spectrum-picker-side-label-quiet-padding-block-start-small` 
 ` --spectrum-picker-side-label-quiet-padding-block-start-medium`
  `--spectrum-picker-side-label-quiet-padding-block-start-large`
  `--spectrum-picker-side-label-quiet-padding-block-start-xlarge`
  
  #### Potential Token Changes: 
  - add component specific tokens for each of the custom properties, including different values for Large scale
  
  
## How and where has this been tested?

Please tag yourself on the tests you've marked complete to confirm the tests have been run by someone other than the author.

### Regression testing

Validate:

1. The documentation pages for at least two other components are still loading, including:

- [ ] The pages render correctly, are accessible, and are responsive.

2. If components have been modified, VRTs have been run on this branch:

- [ ] VRTs have been run and looked at.
- [ ] Any VRT changes have been accepted (by reviewer and/or PR author), or there are no changes.

## Screenshots

Screenshot 2023-11-22 at 11 51 37 AM

## To-do list

- [x] I have read the [contribution guidelines](/.github/CONTRIBUTING.md).
- [ ] I have updated relevant storybook stories and templates.
- [ ] I have tested these changes in Windows High Contrast mode.
- [ ] If my change impacts **other components**, I have tested to make sure they don't break.
- [ ] If my change impacts **documentation**, I have updated the documentation accordingly.
- [ ] ✨ This pull request is ready to merge. ✨

### CODE REVIEW

File: components/picker/metadata/picker.yml [hunk 1]
```diff
@@ -49,6 +49,113 @@ sections:
...
+              <svg class="spectrum-Icon spectrum-UIIcon-Checkmark75 spectrum-Checkbox-checkmark" focusable="false" aria-hidden="true">
+                <use xlink:href="#spectrum-css-icon-Checkmark75" />
+              </svg>
+              <svg class="spectrum-Icon spectrum-UIIcon-Dash75 spectrum-Checkbox-partialCheckmark" focusable="false" aria-hidden="true">
+                <use xlink:href="#spectrum-css-icon-Dash75" />
+              </svg>
+            </span>
+            <span class="spectrum-Checkbox-label">Checkbox</span>
+          </label>
+      </div>
```
Comment (jenndiaz) [inline components/picker/metadata/picker.yml:158]: Adding these examples here for testing purposes should not be merged with these in place

File: tokens/custom-spectrum/custom-large-vars.css [hunk 2]
```diff
@@ -120,4 +120,14 @@ governing permissions and limitations under the License.
...
+
+  --spectrum-picker-side-label-quiet-padding-block-start-small: -1px;
+  --spectrum-picker-side-label-quiet-padding-block-start-medium: -1px;
+  --spectrum-picker-side-label-quiet-padding-block-start-large: -1px;
+  --spectrum-picker-side-label-quiet-padding-block-start-xlarge: 0px;
+
+  --spectrum-fieldlabel-picker-side-label-quiet-padding-block-start-small: 5px;
+  --spectrum-fieldlabel-picker-side-label-quiet-padding-block-start-medium: 11px;
+  --spectrum-fieldlabel-picker-side-label-quiet-padding-block-start-large: 14px;
+  --spectrum-fieldlabel-picker-side-label-quiet-padding-block-start-xlarge: 18px;
```
Comment (mdt2) [inline tokens/custom-spectrum/custom-large-vars.css:132]: Are these values coming from the design team? The numbers seem a bit magical ✨

Comment (jenndiaz) [reply tokens/custom-spectrum/custom-large-vars.css:132]: these are a big magical, This is adjusting each of these, the quiet picker and quiet field label, to be inline with adjacent components (this case this switch). I am awaiting designs thoughts on these values and if they should be tokens as well.

Comment (mdt2) [reply tokens/custom-spectrum/custom-large-vars.css:132]: I'm also a little curious how these look on retina vs non-retina displays. Will things still be aligned?

Comment (jenndiaz) [reply tokens/custom-spectrum/custom-large-vars.css:132]: I did look into this, the macbook I work on does have a built in retina display while my external monitor is an IPS display. 

On the retina monitor (red underline) there is about a half pixel difference 

Screenshot 2023-12-11 at 10 46 11 AM

**Comment by jawinn** [comment]
In [Storybook](https://pr-2316--spectrum-css.netlify.app/preview/?path=/story/components-picker--quiet-side-label&args=size:xl), I'm seeing the medium and extra-large picker text rendering 1px lower than the label text. But it looks fine on the docs example. Is this a result of irregular font rendering within the line-height like we've seen previously? Are the 0px and 1px adjustments trying to compensate for this?

I'm also wondering if we should change this line on picker `margin-block-start: calc( var(--mod-picker-spacing-label-to-picker-quiet, var(--spectrum-picker-spacing-label-to-picker-quiet)) + (1px));`; the mod doesn't let us escape the 1px calc and where is the 1px coming from (possibly should be border-width?).

**Pull request closed without merging**